Lib/http/server.py

```diff
@@ -686,12 +687,42 @@ def send_head(self):
         except OSError:
             self.send_error(HTTPStatus.NOT_FOUND, "File not found")
             return None
+
         try:
+            fs = os.fstat(f.fileno())
+            # Use browser cache if possible
+            if ("If-Modified-Since" in self.headers
+                    and "If-None-Match" not in self.headers):
+                # compare If-Modified-Since and time of last file modification
+                try:
+                    ims = email.utils.parsedate_to_datetime(
+                        self.headers["If-Modified-Since"])
+                except:
```

Comment:
I added the `try` block because you said `parsedate_to_datetime()` could raise exceptions ; do you know which ones ? By looking at the code I see that it might raise `TypeError` because `_parsedate_tz` may return `None` and the line
```python
*dtuple, tz = _parsedate_tz(data)
```
would raise a `TypeError` in this case ; but is it the only one ?

Anyway, the code is itself inside a `try...except` block where the `except` block is just `except:`, so this is not making things much worse.